.github/workflows/codeql-analysis.yml

```diff
@@ -1,66 +1,41 @@
-# For most projects, this workflow file will not need changing; you simply need
-# to commit it to your repository.
-#
-# You may wish to alter this file to override the set of languages analyzed,
-# or to provide custom queries or build logic.
 name: "CodeQL"
 
 on:
   push:
-    branches: [main]
+    branches: [ "main" ]
   pull_request:
-    # The branches below must be a subset of the branches above
-    branches: [main]
+    branches: [ "main" ]
   schedule:
     - cron: '0 15 * * 3'
 
 jobs:
   analyze:
     name: Analyze
-    runs-on: ubuntu-22.04
+
+    runs-on: [ubuntu-24.04]
+    permissions:
+      security-events: write
+      packages: read
+      actions: read
+      contents: read
 
     strategy:
       fail-fast: false
       matrix:
-        # Override automatic language detection by changing the below list
-        # Supported options are ['csharp', 'cpp', 'go', 'java', 'javascript', 'python']
-        language: ['python', 'javascript', 'actions']
-        # Learn more...
-        # https://docs.github.com/en/github/finding-security-vulnerabilities-and-errors-in-your-code/configuring-code-scanning#overriding-automatic-language-detection
+        language: [ "javascript", "python", "actions"]
```

Comment:
Would we want to include the `build-mode` here like in the other PRs as well? Might still be okay but wanted to double-check

Reply:
i have implemented the changes. So currently it will fail on the 'code scanning results / CodeQL'. This is because the language configuration is currently different in Main. I believe it will resolve once this is PR is merged in Runner.